### PR TITLE
[Core][LevelSet] Porting non-sensitive changes from #8519

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**Description**
This PR ports the non-sensitive changes from #8519 (considering non-sensitive changes the addition of new features with no affectation to the current behavior).

The point of this auxiliary PR is to separate the addition of new features the changes in the level set convection algorithm, that is a very sensitive operation used by many people. This would ease the eventual error backtracking.

This needs to be merged prior to #8519.

**Changelog**
- Level set convection element with algebraic stabilization and limiter.

### CODE REVIEW

File: kratos/includes/convection_diffusion_settings.h [hunk 1]
```diff
@@ -220,6 +223,20 @@ class ConvectionDiffusionSettings
...
+        mis_defined_GradientVar=true;
+    }
+    const Variable<array_1d<double,3> >& GetGradientVariable()
+    {
+        return *mpGradientVar;
+    }
+    bool IsDefinedGradientVariable()
+    {
+        return mis_defined_GradientVar;
+    }
```
Comment (philbucher) [inline kratos/includes/convection_diffusion_settings.h]:
```suggestion
    const Variable<array_1d<double,3> >& GetGradientVariable() const
    {
        return *mpGradientVar;
    }
    bool IsDefinedGradientVariable() const
    {
        return mis_defined_GradientVar;
    }
```

File: kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.h [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+#include "includes/variables.h"
+#include "includes/serializer.h"
+#include "includes/cfd_variables.h"
+#include "includes/convection_diffusion_settings.h"
+#include "elements/levelset_convection_element_simplex.h"
+#include "utilities/geometry_utilities.h"
```
Comment (philbucher) [inline kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.h]: some of those can be removed / moved to the cpp

File: kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.cpp [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        return Element::Pointer(new LevelSetConvectionElementSimplexAlgebraicStabilization(NewId, this->GetGeometry().Create(ThisNodes), pProperties));
+        KRATOS_CATCH("");
+    }
+
+    template< unsigned int TDim, unsigned int TNumNodes>
+    void LevelSetConvectionElementSimplexAlgebraicStabilization<TDim, TNumNodes>::CalculateLocalSystem(
+        MatrixType &rLeftHandSideMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo)
+    {
```
Comment (philbucher) [inline kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.cpp:66]: this is a pretty large fct, consider splitting it up

File: applications/ConvectionDiffusionApplication/python_scripts/convection_diffusion_solver.py [hunk 4]
```diff
@@ -414,6 +420,9 @@ def _check_variable_to_set(self, var):
         if (thermal_settings.IsDefinedConvectionVariable()):
             if (thermal_settings.GetConvectionVariable() == var):
                 return True
+        if (thermal_settings.IsDefinedGradientVariable()):
+            if (thermal_settings.GetGradientVariable() == var):
```
Comment (philbucher) [inline applications/ConvectionDiffusionApplication/python_scripts/convection_diffusion_solver.py]:
```suggestion
        if thermal_settings.IsDefinedGradientVariable():
            if thermal_settings.GetGradientVariable() == var:
```

more pythonic

File: applications/ConvectionDiffusionApplication/python_scripts/convection_diffusion_solver.py [hunk 5]
```diff
@@ -167,6 +168,9 @@ def AddVariables(self, target_model_part=None):
         convection_variable = self.settings["convection_diffusion_variables"]["convection_variable"].GetString()
         if (convection_variable != ""):
             convention_diffusion_settings.SetConvectionVariable(KratosMultiphysics.KratosGlobals.GetVariable(convection_variable))
+        gradient_variable = self.settings["convection_diffusion_variables"]["gradient_variable"].GetString()
+        if (gradient_variable != ""):
```
Comment (philbucher) [inline applications/ConvectionDiffusionApplication/python_scripts/convection_diffusion_solver.py]:
```suggestion
        if gradient_variable != "":
```

**Review by philbucher — COMMENTED** [review]
minor comments

File: kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.cpp (hunk 3 above)
Comment (rubenzorrilla) [reply kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.cpp:66]: Having a look to the formulation, I think that right now it is way easy to follow. Considering that the function is around 150 lines (and that good practices say around 100) I'd keep it as it is for the sake of readability.

Maybe adding some inline comments can clarify what is done there.

File: kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.h (hunk 2 above)
Comment (mrhashemi) [reply kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.h]: Including "levelset_convection_element_simplex.h", all these includes are now removed.

**Review by philbucher — APPROVED** [review]

**Pull request merged**